src/ol_infrastructure/applications/learn_ai/__main__.py

```diff
@@ -665,19 +616,19 @@
             dest_secret_labels=k8s_global_labels,
             dest_secret_name=db_creds_secret_name,
             labels=k8s_global_labels,
-            mount=learn_ai_db_vault_backend_config.mount_point,
+            mount=ol_app_db_service.app_db_vault_backend_config.mount_point,
```

Comment:
I would change the variable mounting to be using the generated Vault object rather than the config element. Generally speaking, in the component resources we should be consuming the config models, but not resurfacing them from the instantiated component. Attributes of the component should reference the materialized resources and their attributes.